docs/plugin-types/circuit-executor.md

```diff
@@ -0,0 +1,142 @@
+:::
+
+
+Required Tags:
+
+| Tag                   | Description |
+|-----------------------|-------------|
+| `circuit-executor`    | The identifying tag for plugins implementing this interface.     |
+
+Optional Tags:
+
+| Tag               | Description |
+|:------------------|:------------|
+| `qc-simulator`    | If present, the plugin uses a quantum computing simulator.      |
+| `qiskit`          | A tag specifying the (main) SDK being used by the plugin.      |
+| `qasm` `qasm-2`   | The language that this plugin can understand. The tag `qasm` (for OpenQASM) should always be accompanied by the specific version of qasm supported.     |
+
+
+
+## Interface
+
+```{contents}
+:depth: 2
+:local: True
+```
+
+
+### Main Interaction
+
+The main input of the plugin is therefore such a circuit along with specific execution options.
+Both must be provided in the starting step of the plugin flow, but the execution options are optional.
+The plugins can make some execution options directly available as parameters for the user.
+However, all these parameters must be optional so that the circuit executors can also be used in more automated settings.
+
+Some circuit executor plugins may require authentication credentials in order to access the actual quantum execution resource.
+Those plugins can expose the authentication form as part of the starting step as optional inputs.
+If the credentials are not provided in the first step, a the plugin should ask for the credentials again in a substep form.
+
+:::{hint}
+The circuit executor plugin **must** execute the circuit **as is**, without adding or removing any measurements.
+Transpiling the circuit to an equivalent circuit is allowed.
+:::
+
+
+### Inputs
+
+Circuit executor plugins can have two types of data (i.e. file based) inputs:
+
+ *  **Quantum Circuit** (parameter: `circuit`):\
+    The data input **must** have the data-type [`executable/circuit`](../data-formats/examples/executables.rst#executable-circuit).
+ *  **Execution Options** (parameter `executionOptions`):\
+    The data input **must** be available, but may not receive a value (i.e., this input should be treated as **optional**).
+    The data input **must** have the data-type [`provenance/execution-options`](../data-formats/examples/provenance.rst#provenance-execution-options).
+
+    The execution options **must** support `shots` as the number of shots to simulate/execute.
+    Unknown execution options must be ignored.
+    Execution options that are specific to specific pligins should have unique names.
+
+    | Attribute | Description                                      |
+    |:----------|:-------------------------------------------------|
+    | `shots`   | The number of shots to run the circuit.          |
+
+
+Circuit executor plugins can have two types of direct parameter inputs:
+
+ *  **Execution optiones**: circuit executor plugins can expose some of the execution options as parameters.
```

Comment:
optiones/options. 
Also: This section is hard to understand - does that mean the direct parameter inputs "execution options" override the file-input-based "execution options"?

Reply:
I have changed the text to make this more clear